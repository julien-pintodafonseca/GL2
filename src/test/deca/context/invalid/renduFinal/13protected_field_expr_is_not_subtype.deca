// Description:
//      Utilisation d'un champ protégé : incompatibilité au niveau du typage
//      => Le type de l'expression doit être un sous-type de la classe courante
//
// Resultats:
//      Erreur contextuelle
//      Ligne 15 : le type de a (A) n’est pas un sous-type de B

class A {
    protected int x;
}

class B extends A {
    int getX(A a) {
        return a.x;
    }
}

class X {
    void m() {
        A a = new A();
        B b = new B();
        println(b.getX(a));
    }
}
