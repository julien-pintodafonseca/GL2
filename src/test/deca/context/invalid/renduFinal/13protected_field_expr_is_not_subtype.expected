src\test\deca\context\invalid\renduFinal\13protected_field_expr_is_not_subtype.deca:15:16: Vous essayez d'accéder à un attribut protégé. Or, le type A n'est pas un sous-type de B.
